lib/text/ui_text_displayer.js

```diff
@@ -303,6 +319,16 @@ shaka.text.UITextDisplayer = class {
         shaka.util.Dom.removeAllChildren(this.textContainer_);
         this.currentCuesMap_.clear();
       }
+      if (this.regionElements_.size > 0) {
+        // Clear away any existing regions.
+        for (const regionElement of this.regionElements_.values()) {
```

Comment:
Region elements are put into the text container, and we only make region elements when making a cue element to put into the DOM (so no situation where there's a region element but no cue element), so wouldn't the `removeAllChildren` call above clear away the regions too?

Reply:
Yes, that's true.  Perhaps removeAllChildren() should just be hoisted up to outside of checks on the map sizes, then.  It doesn't really matter if the map is empty or not if we're wiping out the container completely.

Reply:
Cleaned up and simplified.